Standardize (but also add consistent) headers on the admin

Ridhwana opened the pull request:
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Change some h2's to h1's where necessary and add some headers (this wasn't outlined in the task but seemed like a nice to have) to pages where necessary for consistency across the admin.

## Related Tickets & Documents

<!--
For pull requests that relate or close an issue, please include them
below.  We like to follow [Github's guidance on linking issues to pull requests](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

For example having the text: "closes #1234" would connect the current pull
request to issue 1234.  And when we merge the pull request, Github will
automatically close the issue.
-->

- Related Issue #https://github.com/forem/forem/issues/18875
- Closes #https://github.com/forem/forem/issues/18875

## QA Instructions, Screenshots, Recordings

Go through /admin

### UI accessibility concerns?

Improves accessibility with more consistent headings

## Added/updated tests?

- [ ] Yes
- [x] No, and this is why: html changes
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [x] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](gif_link)

Ridhwana commented on Gemfile.lock:
```diff
@@ -739,17 +739,17 @@ GEM
       json-schema (~> 2.2)
       railties (>= 3.1, < 7.1)
-    rubocop (1.41.0)
+    rubocop (1.42.0)
```
Created a [separate PR](https://github.com/forem/forem/pull/18935) with the rubocop update but pulled that branch in because i these files were all erroring.

gadkins commented:
@Ridhwana [This preview job](https://github.com/forem/forem/actions/runs/3891202897/jobs/6641154991#step:16:22) failed because we didn't have forem on the right Uffizzi plan. Your project is now upgraded and will have unlimited previews.

Ridhwana commented:
@rt4914 this is ready for your review as I think you have the most context.

rt4914 commented on app/views/admin/tags/new.html.erb:
```diff
@@ -1,4 +1,4 @@
 <div class="crayons-card p-6">
-  <h4 class="mb-4">New Tag</h4>
+  <h1 class="crayons-title ml-3 xl:ml-0 mb-3">New Tag</h1>
```
The result of this change is 

<img width="500" alt="Screenshot 2023-01-13 at 3 59 08 PM" src="https://user-images.githubusercontent.com/9396084/212298399-e3ce42c9-c25b-4ed6-8d63-25eae2396143.png">

i.e, the `New Tag` text is inside the card.

But if we look at similar other cases like `Create Badge` or `Make a New HTML Variant` the heading is outside the card.
Lets make it consistent.
<img width="300" alt="Screenshot 2023-01-13 at 3 59 02 PM" src="https://user-images.githubusercontent.com/9396084/212298655-2f81520c-68f4-44cd-a879-a751e250c87e.png"> <img width="300" alt="Screenshot 2023-01-13 at 3 59 28 PM" src="https://user-images.githubusercontent.com/9396084/212298679-a51821f9-9b87-4030-a7f8-72475fd5b948.png">

Ridhwana replied:
I put this in a different comment as well. I think the general way that we're moving towards this is that the header + button should be within the card. I say this because Member Manager was the most recently updated page where we got a design from Anuj and the headers were included in the card. However, I think I'd prefer that we get more specific guidance from @anujbhavsar96 on what the best practice is before making changes like this. Perhaps this is also an issue that we can add to our open source board for community contributions in the future.

anujbhavsar96 replied:
@Ridhwana, you are right. All the admin feature we have worked on has header + btn inside the card. That gives a better flow of reading and association with the content below.

IMO, it's not a major issue that affects UX, we can probably revisit these whenever we decide to work on those particular pages holistically, which will help us make better decisions at that time.

rt4914 replied:
Sounds good.

rt4914 commented on app/views/admin/gdpr_delete_requests/index.html.erb:
```diff
@@ -1,8 +1,8 @@
   <header class="flex flex-col l:flex-row justify-content-between l:items-center p-0 m:p-7 pb-4">
-    <h1 class="crayons-title ml-3 l:ml-0">Members<span class="screen-reader-only"> (GDPR Delete Requests)</span></h1>
-    <%= form_with url: admin_gdpr_delete_requests_path, method: :get, local: true, class: "grow-1 mb-3 m:mb-0" do |f| %>
+   <h1 class="crayons-title ml-3 l:ml-0">Members<span class="screen-reader-only"> (GDPR Delete Requests)</span></h1>
```
We have changed the text for to `Invited Members` earlier, so similar to that should this text be changed to `GDPR Actions` ?

<img width="462" alt="Screenshot 2023-01-13 at 3 58 05 PM" src="https://user-images.githubusercontent.com/9396084/212299259-135561a8-05de-4204-9419-56c2c74dd416.png">

Ridhwana replied:
Makes sense to em for them to be the same, I'll update this.

Ridhwana replied:
Looking at this again, perhaps its not so intuitive to make this change. The table is still a table of Members and not only GDPR actions so perhaps it should read Members (GDPR Actions) or maybe I should not have updated `Members` to `Invited Members`  on the previous page. I was trying to align the nav bar with the heading of the page but since @anujbhavsar96 had put a lot of thought and effort into the Member Manager, I'd like to get his thoughts on this as well?

anujbhavsar96 replied:
The table's title should read the same as on the left sidebar. Didn't we have this that way only?

Let's say for Invited Members page title should be `Invited Members,` and for GDPR Actions it should be `GDPR Actions`

Keeping the same title for all three tables looks misleading.

rt4914 replied:
Sounds good.

rt4914 reviewed (changes requested):
Couple of more optional suggestions

1. In mobile screen the left-margin is inconsistent for various headings, see below screenshots
<img width="250" alt="Screenshot 2023-01-13 at 3 53 15 PM" src="https://user-images.githubusercontent.com/9396084/212299595-fe83ff45-5638-4879-8ef1-ec0457daf082.png"> <img width="250" alt="Screenshot 2023-01-13 at 3 53 25 PM" src="https://user-images.githubusercontent.com/9396084/212299612-c9c5a2e5-64dd-4859-a0b2-4e10fc222337.png">

2. The header + button is generally outside of the crayons-card especially in `apps` section, so maybe if needed this can be made consistent too. (Notice if the header + button is inside or outside of crayons-card in screenshots)
<img width="944" alt="Screenshot 2023-01-13 at 4 03 42 PM" src="https://user-images.githubusercontent.com/9396084/212299939-9574d262-e68d-4a47-af61-e2d7eadaa3a8.png">
<img width="910" alt="Screenshot 2023-01-13 at 4 03 47 PM" src="https://user-images.githubusercontent.com/9396084/212299954-c6f5609d-2e6c-4d36-9fb6-cdda2ab48770.png">
<img width="934" alt="Screenshot 2023-01-13 at 4 03 51 PM" src="https://user-images.githubusercontent.com/9396084/212299968-ba7bd884-6f50-4425-9108-3e90e36711cf.png">

Both of these can be considered as separate issues (not related to this PR) too.

Ridhwana commented:
> 1\. In mobile screen the left-margin is inconsistent for various headings, see below screenshots

I'll update this thanks!

> 2\. The header + button is generally outside of the crayons-card especially in `apps` section, so maybe if needed this can be made consistent too. (Notice if the header + button is inside or outside of crayons-card in screenshots)

I think the general way that we're moving towards this is that the header + button should be within the card. I say this because Member Manager was the most recently updated page where we got a design from Anuj and the headers were included in the card. However, I think I'd prefer if we get more specific guidance from @anujbhavsar96 on what the best practice is before making changes like this. Perhaps this is an issue that we can add to our open source board for community contributions in the future.

fdocr reviewed (approved):
Aside from the ongoing header position conversation inline, I tried this out locally and LGTM 🙌🏼

Ridhwana commented:
@rt4914 this is ready for review again - I've fixed the spacing on the headers and renamed the GDPR Actions header. I've left comments on some of the other issues and I think that we can fix then as we improve on the designs on the admin pages.

Ridhwana commented:
Mistakenly committed another feature  👆 to this branch, hence some of these commits may not make sense but I've removed it.
